### ISSUE TITLE
Argument exception

### ISSUE THREAD

**Issue by xplicit** [davidfowl/HelloWorldVNext#1]
when I run `k web`, I get an exception
OS Ubuntu linux 12.04

```
mono --version
Mono JIT compiler version 3.4.1 (master/09126b1 Mon May 26 03:54:44 NOVT 2014)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
    TLS:           __thread
SIGSEGV:       altstack
Notifications: epoll
Architecture:  x86
Disabled:      none
Misc:          softdebug 
LLVM:          supported, not enabled.
GC:            sgen
```

k web

System.ArgumentException: An element with the same key already exists in the dictionary.
  at System.Collections.Generic.Dictionary`2[System.String,System.String].Add (System.String key, System.String value) [0x00000] in <filename unknown>:0 
  at Microsoft.Framework.Runtime.FrameworkReferenceResolver.PopulateCache () [0x00000] in <filename unknown>:0 
  at Microsoft.Framework.Runtime.FrameworkReferenceResolver..ctor () [0x00000] in <filename unknown>:0 
  at Microsoft.Framework.Runtime.ReferenceAssemblyDependencyResolver..ctor () [0x00000] in <filename unknown>:0 
  at Microsoft.Framework.Runtime.DefaultHost.Initialize (Microsoft.Framework.Runtime.DefaultHostOptions options) [0x00000] in <filename unknown>:0 
  at Microsoft.Framework.Runtime.DefaultHost..ctor (Microsoft.Framework.Runtime.DefaultHostOptions options, IServiceProvider hostProvider) [0x00000] in <filename unknown>:0 
  at Microsoft.Framework.ApplicationHost.Program.Main (System.String[] args) [0x00000] in <filename unknown>:0 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00054] in /home/sergey/Projects/mono/mono/mcs/class/corlib/System.Reflection/MonoMethod.cs:230

**Comment by davidfowl** [comment]
Hmmm. Do you get the same if you k run?

**Comment by davidfowl** [comment]
Filed a bug

**Comment by davidfowl** [comment]
Should be fixed now. Run kvm upgrade and try again.

**Comment by xplicit** [comment]
Thanks, it's working now, but performance is very slow, I am going to add new issue.

**Comment by davidfowl** [comment]
You can't host anything production quality on this stack yet. No performance work as been done as yet. There's probably also a can of worms we need to figure out on mono so sit tight and help find issues :smile:

**Comment by xplicit** [comment]
OK :) by the way I'll add some info for you. I am developing fastcgi web server for mono and did alot of mono profiling while developing to improve server performance and  would like to share the information I've got. I've written several posts about it at http://forcedtoadmin.blogspot.com and currently I am at the end of moving my fastcgi server from managed listener to libevent (I did it already, just need to document configuration settings). So I put some info in the issue #2
